### PR TITLE
[GEN-666] Sets Created (Custom), and Filtering By Set

### PR DESCRIPTION
**Pull request by abhipi**
- Sets can now be created within Climbs (Old ones, and Custom names- makes a new Set Item).
- Sets are updated alongside climbs (deletion, updated set)- COMPLETELY FLEXIBLE.
- Filtering on the Live Climb Tracker is now done by set too.
WhatsApp Image 2024-02-05 at 1 10 53 AM
WhatsApp Image 2024-02-05 at 1 10 53 AM (1)

### CODE REVIEW

**Review by VGmt — APPROVED** [review]

File: trk/src/Screens/TabScreens/ClimbCreate/Frontend/index.js [hunk 1]
```diff
@@ -30,17 +31,29 @@ const ClimbInputData = (props) => {
...
+          setSelectedValue(climbData.set); // Additional state update for selected value, if necessary
+        }
+        setIfsc(climbData.ifsc);
+        setInfo(climbData.info);
+      }
+    };
+  
+    fetchData();
+  }, [isEditMode, climbData]); // Dependencies
+
```
Comment (redpepper-nag) [inline trk/src/Screens/TabScreens/ClimbCreate/Frontend/index.js:56]: love you how you dynamically create set ids for new sets added to the dropdown

File: trk/src/Screens/TabScreens/ClimbCreate/Frontend/index.js [hunk 2]
```diff
@@ -89,6 +108,43 @@ const ClimbInputData = (props) => {
...
+          // Update the existing custom item label without removing it
+          return prevItems.map(item => item.value === 'custom' ? { ...item, label: `Add "${searchText}"`, name: searchText.trim()} : item);
+        }
+      } else {
+        // If searchText is empty or there's a match, remove "Add custom" option
+        return prevItems.filter(item => item.value !== 'custom');
+      }
+    });
+  }, [searchText]);  
+
```
Comment (redpepper-nag) [inline trk/src/Screens/TabScreens/ClimbCreate/Frontend/index.js:147]: amazing that you can create new sets on the fly. Well done!

File: trk/src/Screens/TabScreens/ClimbCreate/Frontend/index.js [hunk 3]
```diff
@@ -133,35 +189,92 @@ const ClimbInputData = (props) => {
...
+        } else if ((!climbData || selectedValue !== climbData.set)) { //If not a duplicate, then make a custom set
+          const setObj = {
+            archived: false,
+            name: filteredSets[0].name,
+            climbs: [newClimbId], //replace with newClimbId
+            setters: [currentUser.uid]
+          }
+          await SetsApi().addSet(setObj);
+        }
+
```
Comment (redpepper-nag) [inline trk/src/Screens/TabScreens/ClimbCreate/Frontend/index.js:258]: Is the oldSet a set that is archived or no longer live? or does it refer to something different?

**Review by redpepper-nag — APPROVED** [review]

File: trk/src/Screens/TabScreens/ClimbCreate/Frontend/index.js (hunk 3 above)
Comment (abhipi) [reply trk/src/Screens/TabScreens/ClimbCreate/Frontend/index.js:258]: It is the previous set attached to the Climb. The sets aren't archived (yet), just emptied.

**Pull request merged**